### PR TITLE
Prevent RatingSwitch buttons from stealing focus

### PR DESCRIPTION
**Pull request by 4vanger**
Problem:
1) click + or - button on post/comment
2) press "Space"
3) (even worse) scroll down few screens without clicking using mouse/touchpad
Expected behavior: page should scroll down one screen
Actual behavior: button is clicked again undoing the vote. Even more frustrating is button is not visible on a screen anymore

Fix: unfocus rating buttons after click

### CODE REVIEW

File: frontend/src/Components/RatingSwitch.tsx [hunk 1]
```diff
@@ -89,33 +89,39 @@ export default function RatingSwitch(props: RatingSwitchProps) {
...
-        if (props.onVote) {
-            props.onVote(newRating, vote);
-        }
-
-        api.voteAPI.vote(props.type, props.id, vote)
-            .then(result => {
-                setState({
-                    rating: result.rating,
-                    vote: result.vote
+        return (ev: React.MouseEvent) => {
```
Comment (Aivean) [inline frontend/src/Components/RatingSwitch.tsx:92]: nit: could be simplified as  
`(vote: number) => (ev: React.MouseEvent) => { ... }`

**Review by Aivean — APPROVED** [review]
Checked locally, works as expected.

File: frontend/src/Components/RatingSwitch.tsx (hunk 1 above)
Comment (4vanger) [reply frontend/src/Components/RatingSwitch.tsx:92]: too hard to read to my taste

Comment (Aivean) [reply frontend/src/Components/RatingSwitch.tsx:92]: Dunno what you are talking about, looks like idiomatic Haskell ;p

**Pull request merged**